### ISSUE TITLE
U.S. subdivision flags

### ISSUE THREAD

**Issue by maxterry** [hfg-gmuend/openmoji#467]
As with other emoji sets, looks like only England, Scotland, and Wales are currently included as subdivision flags in OpenMoji:

https://openmoji.org/library/#group=flags%2Fsubdivision-flag

State and territory flags for the U.S. are defined and possible, just not implemented: https://blog.emojipedia.org/us-state-flag-emojis-now-possible

Would be cool to get them included here.

Related to https://github.com/hfg-gmuend/openmoji/issues/416

**Comment by Kirylchyk** [comment]
hey @maxterry, I picked up and started to work on this enhancement. https://github.com/hfg-gmuend/openmoji/pull/489 was merged to add Washington DC flag emoji. I am aware of the following flags https://en.wikipedia.org/wiki/Category:United_States_state_flags and planning to add flags at the reasonable pace. Do you have any order preferences? Thank you

**Comment by xavizardKnight** [comment]
@Kirylchyk @alexcochran Heyas! I want to start advancing with the US subdivision flags, now that the subdivision flags of Spain are done, and I've seen that both of you have contributed some recently. Have you done any more advancements on other flags? Can I continue on working on the remaining flags?

**Comment by Kirylchyk** [comment]
hey @xavizardKnight that's awesome! Indeed, i'm working on some more emoji. Lets split

**Comment by maxterry** [comment]
Nice to see some momentum on this! Would be great to work toward a complete set of U.S. subdivision flags eventually, no preference from my side on order/priority to get there.

**Comment by xavizardKnight** [comment]
> hey @xavizardKnight that's awesome! Indeed, i'm working on some more emoji. Lets split

Yeah! If you started some, give me the list of those you haven't started yet. Or let's split the list 50/50 and start working from there, as you prefer.


> Nice to see some momentum on this! Would be great to work toward a complete set of U.S. subdivision flags eventually, no preference from my side on order/priority to get there.

We can try to aim and see if we can complete all the US subdivision flags before the next OpenMoji release; there are quite a few flags, and some of them are pretty complex, but I think we can manage to pull it off.
Don't know when it will be the next release, but probably by the end of this year.

**Comment by xavizardKnight** [comment]
Also, I've realised that the recently added Colorado flag has no grid layer in its source file. Technically, it does have a grid layer, but it's hidden. This doesn't seem to affect the outputs, but it looks weird in comparison with the rest of the flags when viewing the thumbnails of the folder. Was this intentional?

![2024-04-09_19-43](https://github.com/hfg-gmuend/openmoji/assets/35200790/0f386fd3-182b-4f08-8f4d-f197759fec74)

**Comment by alexcochran** [comment]
The hidden grid was not intentional; i just don't think I noticed it was
different than the others during development.

I'd be happy to work on some more of the US states collaboratively.

**Comment by xavizardKnight** [comment]
> The hidden grid was not intentional; i just don't think I noticed it was different than the others during development. I'd be happy to work on some more of the US states collaboratively.

Don't worry. I'll do now a quick PR readding the grid, it's a quick fix.

> let's split the list 50/50

We can split the remaining list in 3 parts; then all 3 we can contribute.

**Comment by Kirylchyk** [comment]
agree, i have no preferences. 

By now, i'm working on:
 **[Flags of Alabama]**(https://en.wikipedia.org/wiki/Category:Flags_of_Alabama)‎ (1 C, 1 P) 
**[Flags of Utah]**(https://en.wikipedia.org/wiki/Category:Flags_of_Utah)‎ (1 C, 1 P)

And then we can split, what is left

**Comment by Kirylchyk** [comment]
@alexcochran @xavizardKnight fyi opened PR for Alabama flag ->

Reference 1: https://github.com/hfg-gmuend/openmoji/pull/497
Reference 2: https://en.wikipedia.org/wiki/Flag_of_Alabama

Thx!

**Comment by xavizardKnight** [comment]
> alexcochran xavizardKnight fyi opened PR for Alabama flag

Nice!

I've been starting to experiment with the Hawaii, Arizona, Washington (state, not DC), Ohio and Montana flags. They look a bit tricky, and I like challenges, so I'm playing with them to see what I can do.

We really should create a list or something so we can see who does what and what remains…

Also, it would be nice to not create one PR per flag; let's make all of us one PR per person with all the US subdivision flags when we finish them all. Otherwise, there will be a lot of small PRs.

**Comment by Kirylchyk** [comment]
i would prefer to keep each flag per one separate commit

**Comment by b-g** [comment]
Super that the two of you starting to tackle this! Great!

Small note: I would prefer less PRs than for every single flag a dedicated PR. Generating and checking is quite some work. Sorry! OpenMoji is quite heavy ...

**Comment by Kirylchyk** [comment]
Gotcha, i'll try to combine several flags in one PR. I would prefer to work on my own branch at my pace. Thank you

**Comment by xavizardKnight** [comment]
Heyas! Have any of you been progressing on more U.S. flags lately?

I have completed the flags of Arizona, Hawaii, Montana, Ohio and the Washington state.

![image](https://github.com/hfg-gmuend/openmoji/assets/35200790/ddeadef6-0978-44de-a679-a847f63206dc)

*(I know that George Washington himself is not a flag, but I needed an emoji of him for the Washington State flag, that's why he's here xD)*

I'll start tackling more flags soon. How are your progresses going?

**Comment by Kirylchyk** [comment]
hey @xavizardKnight that's awesome! Im still working on a couple of flags

**Comment by xavizardKnight** [comment]
Heyas! Any new progress on new US subdivision flags? I personally haven't advanced much lately (sorry…), but I plan on going back to do more flags soon.

I'll start working on Kansas, Florida, Illinois, Minnesota, Oregon and Wisconsin (hope to not cause any conflicts!).

Also, I compiled a list of all the US flags to see what's done and what not and who's doing what (pls tell me if I forgot or I mistake'd something).

```
Alabama (US-AL): Kirylchyk
Alaska (US-AK)
American Samoa (US-AS)
Arizona (US-AZ): xavizardKnight (finished locally)
Arkansas (US-AR)
California (US-CA): RELEASED
Colorado (US-CO): ON MASTER
Connecticut (US-CT)
Delaware (US-DE)
Florida (US-FL): xavizardKnight
Georgia (US-GA)
Guam (US-GU)
Hawaii (US-HI): xavizardKnight (finished locally)
Idaho (US-ID)
Illinois (US-IL): xavizardKnight
Indiana (US-IN)
Iowa (US-IA)
Kansas (US-KS): xavizardKnight
Kentucky (US-KY)
Louisiana (US-LA)
Maine (US-ME)
Maryland (US-MD)
Massachusetts (US-MA)
Michigan (US-MI)
Minnesota (US-MN): xavizardKnight
Mississippi (US-MS)
Missouri (US-MO)
Montana (US-MT): xavizardKnight (finished locally)
Nebraska (US-NE)
Nevada (US-NV)
New Hampshire (US-NH)
New Jersey (US-NJ)
New Mexico (US-NM)
New York (US-NY)
North Carolina (US-NC)
North Dakota (US-ND)
Northern Mariana Islands (US-MP)
Ohio (US-OH): xavizardKnight (finished locally)
Oklahoma (US-OK)
Oregon (US-OR): xavizardKnight
Pennsylvania (US-PA)
Puerto Rico (US-PR): xavizardKnight (finished locally)
Rhode Island (US-RI)
South Carolina (US-SC)
South Dakota (US-SD)
Tennessee (US-TN)
Texas (US-TX): RELEASED
U.S. Outlying Islands (US-UM): xavizardKnight (finished locally)
U.S. Virgin Islands (US-VI): xavizardKnight (finished locally)
Utah (US-UT): Kirylchyk
Vermont (US-VT)
Virginia (US-VA)
Washington (US-WA): xavizardKnight (finished locally)
Washington DC (US-DC): ON MASTER
West Virginia (US-WV)
Wisconsin (US-WI): xavizardKnight
Wyoming (US-WY)
```

"RELEASED" means that this flag has been released on a stable OpenMoji release (and is searchable on the Library webpage, for example). "ON MASTER" means it's finished and pushed to the master branch, but still hasn't made its way into a stable release. Usernames indicate that a flag is WIP or not yet pushed, and who is making it. Flags with no username and no text behind its name are up-for-grabs.

To update the list, copy the list, edit it and send it as a new message here. Or tell me what flags you've done and I'll update it for you.

Let's see if this way we can speed up the flags that we've left!

**Comment by Kirylchyk** [comment]
@xavizardKnight cool, thx! feel free to pick up mine, am busy with other projects

**Comment by jsskeen-utah** [comment]
One year later, is anyone still actively working on this? Any updates to the list of ones that are available to pick up? I'm working on a project that could greatly benefit from having an emoji for each state flag, or at the very least some sort of placeholder if there isn't one.

**Comment by alexcochran** [comment]
Sorry for going quiet after finishing one flag.

My design skills might be a bit thin for some of the more complex designs. I'm happy to assist with simpler designs/components as I can, but I'd prefer to leave choices up to those confident enough in their vector abilities to make a meaningful run at this.

**Comment by jsskeen-utah** [comment]
SVGs for the different US state flags can be found here: https://en.wikipedia.org/wiki/Flags_of_the_U.S._states_and_territories (Their license is public domain). I would imagine it wouldn't be too hard to take these and massage them however needed for this use case.
